src/components/quantity-counter/QuantitiyCounter.jsx

```diff
@@ -0,0 +1,42 @@
+import AddRoundedIcon from '@mui/icons-material/AddRounded';
+import RemoveRoundedIcon from '@mui/icons-material/RemoveRounded';
+import styles from './QuantitiyCounter.module.css';
+import { useEffect, useState } from 'react';
+
+export const QuantityCounter = ({quantityValue = 0}) => {
+    const [value, setValue] = useState(quantityValue);
+
+    const increaseHandler = () => {
+        setValue(value + 1);
+    }
+
+    const decreaseHandler = () => {
+        setValue(value - 1);
+    }
+
+    useEffect(() => {
+        if (!value) {
+            document.getElementById("dec").style.pointerEvents = "none";
+            document.getElementById("dec").style.opacity = "0.2";
+        } else {
+            document.getElementById("dec").style.pointerEvents = "auto";
+            document.getElementById("dec").style.opacity = "1";
+        }
```

Comment:
don't target the real DOM directly in react this messes up the whole react virtual DOM reactivity system, should do those in a variable then apply it to element's style prop